### ISSUE TITLE
[FEATURE] Single cache for CNMF extensions

### ISSUE THREAD

**Issue by kushalkolar** [nel-lab/mesmerize-core#5]
Create a single cache, probably using a class decorator, for CNMF outputs (contours, the hdf5 file etc.) because they can take a few seconds to load sometimes.

**Comment by kushalkolar** [comment]
Starting point:

```python
from time import sleep
from functools import wraps

class Cache:
    def __init__(self):
        pass
    
    @wraps(func)
    def use_cache(self, *args, **kwargs):
        print(hex(id(self)))
        return

cache = Cache()
 
@cache.use_cache
def get_output(a, b=None):
    sleep(1)
    return a, b

@cache.use_cache
def get_output2(c, d, e=None):
    sleep(1)
    return c, d, e
```

**Comment by kushalkolar** [comment]
@clewis7 an outline idea I had for caching, we can discuss after your finish the tests

**Comment by kushalkolar** [comment]
implemented in #62
